### DISCUSSION TITLE
Using q-tree to make a comments section

### DISCUSSION THREAD

**Discussion by bxia-paul** [quasarframework/quasar#12568] [General - Components / Directives / etc]
I would like to pull data from an API and display it using the q-tree format however, I am not certain how to go about this.
The goal is to accomplish something like this 
![image](https://user-images.githubusercontent.com/92060355/155250506-b8bc7cac-567e-4c8c-834f-a264b247c9fe.png)

I have pulled the data using axios in the composition API. However, I am stuck with how to transfer the data from the request to the tree component. Any information on how to go about completing this task will be helpful. Attached is my comments component.

`<template>`
 ```
 <br>
  <div class="q-pa-md q-gutter-sm">
    <q-tree
      :nodes="tree"
      node-key="label"
    />
  </div>
```
`</template>`

`<script>`

```
import { api } from 'boot/axios'
import { useQuasar } from 'quasar'
import { ref} from 'vue'
import { onMounted} from 'vue'
export default {
```
  
```
  setup () {
      const tree = [
        {
        label: 'Start ' ,
        children : [ ]
      }
      ]
```
      ```
      const $q = useQuasar()
      const data = ref(null)
      const comments = ref([])
      const commTags = ref([])
      const postTags = ref([])
```

      ```
function loadComments(){
         
         let url = "https://swarmnet-staging.herokuapp.com/replies"
         
          api.get(url,{
          method: 'GET',
          
          headers: {
                  'Access-Control-Allow-Origin': '*'
                }
            })
          .then((response) => {
            data.value = response.data
            tree.label = data.value[0].text
           
            console.log(tree.label)
            
            for (let i of data.value) {
                comments.value.push(i)
                commTags.value.push(i.tags) 
     
            } 
          
           console.log( " Comments Tags ")
          
          
          })
          .catch(() => {
            $q.notify({
              color: 'negative',
              position: 'top',
              message: 'Loading failed in comment component',
              icon: 'report_problem'
            })
          })

      }
```

      ```
`onMounted(() => {`
     `  loadComments();`
   ` })`
```

   ```
       return {
        `loadComments,`
       ` comments,`
       ` postTags,`
       ` commTags,`
       ` data,`
        `tree,`
  }
}
```
`</script>`

**Comment by metalsadman** [comment]
closest would be https://quasar.dev/vue-components/tree#lazy-loading
and also read more about its structure https://quasar.dev/vue-components/tree#nodes-model-structure

**Reply by bxia-paul** [reply]
I have tried, lazyload but I just don't know how to allow the API data to be stored in the node. I keep seeing in all examples that the content of the node is hardcoded.

**Comment by MilosPaunovic** [comment]
On the examples Aldrin forwarded you there is a Codepen link which shows lazy-loading nodes using a `setTimeout` to fake server call.
[https://codepen.io/milospaunovic/pen/BamOJog?editors=101](https://codepen.io/milospaunovic/pen/BamOJog?editors=101)

What you need to do is to fetch data from your API and set it to be the value of variable you're using as `nodes` prop in your `QTree`.

If you're still having issues, I suggest that you create a Codepen reproduction with exact problem you're facing, as it will be easier for someone to help you.
